ingest-app/src/cmr/ingest/services/subscriptions_helper.clj

```diff
@@ -201,3 +202,22 @@
                            (remove :deleted)
                            (map #(select-keys % [:concept-id :extra-fields :metadata])))]
     (send-subscription-emails context (process-subscriptions context subscriptions))))
+
+(defn throw-error-if-one-sided-date-map
+  "Throws a service error if the provided map does not contain both keys :start-date and :end-date"
+  [date-map]
+  (when (not-every? date-map [:start-date :end-date])
+    (errors/throw-service-error :bad-request "One-sided date formats are not permitted. Please provide a start and end date."))
+  date-map)
+
+(defn trigger-email-subscription-processing
+  "Trigger subscription processing given the provided revision-date.  Revision date is passed through to
+   a Search query and should follow the same format as the CMR Search API."
+  [context params]
+  (let [revision-date (:revision-date params)]
+    (-> revision-date
+        (date-time-range-parser/parse-datetime-range)
+        (throw-error-if-one-sided-date-map)
```

Comment:
We are missing the validation that the end-date of the revision-date-range must be after the start-date. I would make a separate validation function and call it here rather than using the threading macro. I would also pass the revision-date-range explicitly to the `email-subscription-processing` function rather than throwing it into the `context`.